### DISCUSSION TITLE
best practices for docx to pdf

### DISCUSSION THREAD

**Discussion by tobias-feddersen** [gotenberg/gotenberg#969] [Q&A]
Hello together,

are there any best practices for converting docx to pdf ?

i am struggling a bit and want to know what options do i have to optimize the pdf output.
Normally in Libre Office you can open/import docx with some compatibility settings to improve the document and fix the most common issues.

At the moment i am facing the issue that printed documents have a "Table of Content" that is missing its spacing and some tables have wrong spacing as well. In addition i have some blank pages, which are printed even with setting skipEmptyPages = true, i tried exportPlaceholders and exportFormFields as well, but no difference.

So i hope some of you have some best practices to improve the pdf quality  :)

**Comment by gulien** [comment]
Hello @TobiasB87 👋 

If you found some please share!

**Reply by gulien** [reply]
Not all are implemented, but you can find the parameters I use in Gotenberg: https://help.libreoffice.org/latest/en-US/text/shared/guide/pdf_params.html. AFAIK, cannot do much more alas.

I wonder is there are some hidden tricks but 🤷‍♂️

**Reply by tobias-feddersen** [reply]
ah so Gotenberg is directly triggering the pdf export function of Libre Office. hm so there is not much that can be done.
Looks like i need to find another way to "print" my 20k word documents.
